From 38ddbe64a462bc44aa36efc7736e6e374bb2c4c5 Mon Sep 17 00:00:00 2001
From: sczs <sczs@chromium.org>
Date: Mon, 25 Feb 2019 06:08:12 +0000
Subject: [PATCH] [ios] Cleans up InfobarBannerViewController presentation.

- Deletes InfobarContainerVC since there will be no "stacked"
InfobarBanners for the time being. Keeping this adds unneeded complexity.
- Uses Window to configure the InfobarBanner container and presented views.
- Removes the presentedView on a successful dismissal in InfobarBannerAnimator.

Bug: 911864
Change-Id: I52974895040eadfd8917570fc92d852ee6411f8c
Reviewed-on: https://chromium-review.googlesource.com/c/1480708
Commit-Queue: Sergio Collazos <sczs@chromium.org>
Reviewed-by: Chris Lu <thegreenfrog@chromium.org>
Reviewed-by: Peter Lee <pkl@chromium.org>
Cr-Commit-Position: refs/heads/master@{#635033}
---
 ios/chrome/browser/ui/infobars/BUILD.gn       |  2 -
 .../infobar_password_coordinator.mm           |  1 +
 .../infobars/infobar_container_coordinator.mm | 38 ++++++------------
 .../infobar_container_view_controller.h       | 20 ----------
 .../infobar_container_view_controller.mm      | 39 -------------------
 .../presentation/infobar_banner_animator.h    |  4 +-
 .../presentation/infobar_banner_animator.mm   |  6 +++
 .../infobar_banner_presentation_controller.h  |  3 +-
 .../infobar_banner_presentation_controller.mm | 26 ++++++-------
 9 files changed, 34 insertions(+), 105 deletions(-)
 delete mode 100644 ios/chrome/browser/ui/infobars/infobar_container_view_controller.h
 delete mode 100644 ios/chrome/browser/ui/infobars/infobar_container_view_controller.mm

diff --git a/ios/chrome/browser/ui/infobars/BUILD.gn b/ios/chrome/browser/ui/infobars/BUILD.gn
index 24181ebb67b5d..df279f7dd3fa6 100644
--- a/ios/chrome/browser/ui/infobars/BUILD.gn
+++ b/ios/chrome/browser/ui/infobars/BUILD.gn
@@ -58,8 +58,6 @@ source_set("infobars_ui") {
     "infobar_constants.h",
     "infobar_constants.mm",
     "infobar_container_consumer.h",
-    "infobar_container_view_controller.h",
-    "infobar_container_view_controller.mm",
     "infobar_ui_delegate.h",
     "legacy_infobar_container_view_controller.h",
     "legacy_infobar_container_view_controller.mm",
diff --git a/ios/chrome/browser/ui/infobars/coordinators/infobar_password_coordinator.mm b/ios/chrome/browser/ui/infobars/coordinators/infobar_password_coordinator.mm
index 4a41e4715b2bb..1604741570ac9 100644
--- a/ios/chrome/browser/ui/infobars/coordinators/infobar_password_coordinator.mm
+++ b/ios/chrome/browser/ui/infobars/coordinators/infobar_password_coordinator.mm
@@ -45,6 +45,7 @@ - (instancetype)initWithInfoBarDelegate:
   self = [super initWithBaseViewController:nil browserState:nil];
   if (self) {
     _passwordInfoBarDelegate = passwordInfoBarDelegate;
+    _presented = YES;
   }
   return self;
 }
diff --git a/ios/chrome/browser/ui/infobars/infobar_container_coordinator.mm b/ios/chrome/browser/ui/infobars/infobar_container_coordinator.mm
index 9eb605f0cddf5..4b42324a7a5e8 100644
--- a/ios/chrome/browser/ui/infobars/infobar_container_coordinator.mm
+++ b/ios/chrome/browser/ui/infobars/infobar_container_coordinator.mm
@@ -13,7 +13,6 @@
 #import "ios/chrome/browser/ui/infobars/coordinators/infobar_coordinating.h"
 #import "ios/chrome/browser/ui/infobars/infobar_container_consumer.h"
 #include "ios/chrome/browser/ui/infobars/infobar_container_mediator.h"
-#include "ios/chrome/browser/ui/infobars/infobar_container_view_controller.h"
 #import "ios/chrome/browser/ui/infobars/infobar_feature.h"
 #import "ios/chrome/browser/ui/infobars/infobar_positioner.h"
 #include "ios/chrome/browser/ui/infobars/legacy_infobar_container_view_controller.h"
@@ -33,9 +32,8 @@ @interface InfobarContainerCoordinator () <
 
 @property(nonatomic, assign) TabModel* tabModel;
 
-// UIViewController that contains Infobars.
-@property(nonatomic, strong)
-    InfobarContainerViewController* containerViewController;
+// ViewController of the Infobar currently being presented, can be nil.
+@property(nonatomic, weak) UIViewController* infobarViewController;
 // UIViewController that contains legacy Infobars.
 @property(nonatomic, strong)
     LegacyInfobarContainerViewController* legacyContainerViewController;
@@ -62,11 +60,6 @@ - (void)start {
   DCHECK(self.positioner);
   DCHECK(self.dispatcher);
 
-  // Creates the InfobarContainerVC.
-  InfobarContainerViewController* container =
-      [[InfobarContainerViewController alloc] init];
-  self.containerViewController = container;
-
   // Creates the LegacyInfobarContainerVC.
   LegacyInfobarContainerViewController* legacyContainer =
       [[LegacyInfobarContainerViewController alloc]
@@ -105,7 +98,7 @@ - (void)stop {
 
 - (void)hideContainer:(BOOL)hidden {
   [self.legacyContainerViewController.view setHidden:hidden];
-  [self.containerViewController.view setHidden:hidden];
+  [self.infobarViewController.view setHidden:hidden];
 }
 
 - (UIView*)legacyContainerView {
@@ -135,27 +128,20 @@ - (void)addInfoBarWithDelegate:(id<InfobarUIDelegate>)infoBarDelegate
   ChromeCoordinator<InfobarCoordinating>* infobarCoordinator =
       static_cast<ChromeCoordinator<InfobarCoordinating>*>(infoBarDelegate);
 
+  // Present the InfobarCoordinator BannerViewController.
   [infobarCoordinator start];
-
-  // Add the infobarCoordinator bannerVC to the containerVC.
-  InfobarContainerViewController* containerViewController =
-      base::mac::ObjCCastStrict<InfobarContainerViewController>(
-          self.containerViewController);
-  [containerViewController
-      addInfobarViewController:static_cast<UIViewController*>(
-                                   [infobarCoordinator bannerViewController])];
-
-  // Present the containerVC.
-  self.containerViewController.transitioningDelegate = self;
-  [self.containerViewController
+  self.infobarViewController = [infobarCoordinator bannerViewController];
+  [infobarCoordinator bannerViewController].transitioningDelegate = self;
+  [[infobarCoordinator bannerViewController]
       setModalPresentationStyle:UIModalPresentationCustom];
-  [self.baseViewController presentViewController:self.containerViewController
-                                        animated:YES
-                                      completion:nil];
+  [self.baseViewController
+      presentViewController:[infobarCoordinator bannerViewController]
+                   animated:YES
+                 completion:nil];
 }
 
 - (void)setUserInteractionEnabled:(BOOL)enabled {
-  [self.containerViewController.view setUserInteractionEnabled:enabled];
+  [self.infobarViewController.view setUserInteractionEnabled:enabled];
 }
 
 - (void)updateLayoutAnimated:(BOOL)animated {
diff --git a/ios/chrome/browser/ui/infobars/infobar_container_view_controller.h b/ios/chrome/browser/ui/infobars/infobar_container_view_controller.h
deleted file mode 100644
index 0031c7f780db7..0000000000000
--- a/ios/chrome/browser/ui/infobars/infobar_container_view_controller.h
+++ /dev/null
@@ -1,20 +0,0 @@
-// Copyright 2018 The Chromium Authors. All rights reserved.
-// Use of this source code is governed by a BSD-style license that can be
-// found in the LICENSE file.
-
-#ifndef IOS_CHROME_BROWSER_UI_INFOBARS_INFOBAR_CONTAINER_VIEW_CONTROLLER_H_
-#define IOS_CHROME_BROWSER_UI_INFOBARS_INFOBAR_CONTAINER_VIEW_CONTROLLER_H_
-
-#import <UIKit/UIKit.h>
-
-
-// TODO(crbug.com/1372916): PLACEHOLDER Work in Progress class for the new
-// InfobarUI. ViewController that contains all Infobars.
-@interface InfobarContainerViewController : UIViewController
-
-// Adds a new InfobarViewController to this container.
-- (void)addInfobarViewController:(UIViewController*)infobarViewController;
-
-@end
-
-#endif  // IOS_CHROME_BROWSER_UI_INFOBARS_INFOBAR_CONTAINER_VIEW_CONTROLLER_H_
diff --git a/ios/chrome/browser/ui/infobars/infobar_container_view_controller.mm b/ios/chrome/browser/ui/infobars/infobar_container_view_controller.mm
deleted file mode 100644
index 1b640c51cf6d1..0000000000000
--- a/ios/chrome/browser/ui/infobars/infobar_container_view_controller.mm
+++ /dev/null
@@ -1,39 +0,0 @@
-// Copyright 2018 The Chromium Authors. All rights reserved.
-// Use of this source code is governed by a BSD-style license that can be
-// found in the LICENSE file.
-
-#import "ios/chrome/browser/ui/infobars/infobar_container_view_controller.h"
-
-#if !defined(__has_feature) || !__has_feature(objc_arc)
-#error "This file requires ARC support."
-#endif
-
-// TODO(crbug.com/1372916): PLACEHOLDER Work in Progress class for the new
-// InfobarUI.
-@implementation InfobarContainerViewController
-
-#pragma mark - UIViewController
-
-- (void)viewDidLoad {
-  self.view = [[UIView alloc] initWithFrame:CGRectZero];
-}
-
-- (void)addInfobarViewController:(UIViewController*)infobarViewController {
-  [self addChildViewController:infobarViewController];
-  [self.view addSubview:infobarViewController.view];
-  [infobarViewController didMoveToParentViewController:self];
-  infobarViewController.view.translatesAutoresizingMaskIntoConstraints = NO;
-  [NSLayoutConstraint activateConstraints:@[
-    [infobarViewController.view.topAnchor
-        constraintEqualToAnchor:self.view.topAnchor
-                       constant:10],
-    [infobarViewController.view.bottomAnchor
-        constraintEqualToAnchor:self.view.bottomAnchor],
-    [infobarViewController.view.leadingAnchor
-        constraintEqualToAnchor:self.view.leadingAnchor],
-    [infobarViewController.view.trailingAnchor
-        constraintEqualToAnchor:self.view.trailingAnchor]
-  ]];
-}
-
-@end
diff --git a/ios/chrome/browser/ui/infobars/presentation/infobar_banner_animator.h b/ios/chrome/browser/ui/infobars/presentation/infobar_banner_animator.h
index 96fd220316cff..2c4fd1cf32595 100644
--- a/ios/chrome/browser/ui/infobars/presentation/infobar_banner_animator.h
+++ b/ios/chrome/browser/ui/infobars/presentation/infobar_banner_animator.h
@@ -7,8 +7,8 @@
 
 #import <UIKit/UIKit.h>
 
-// TODO(crbug.com/1372916): PLACEHOLDER Work in Progress class for the new
-// InfobarUI.
+// Animator used to present an InfobarBanner dropping from the top of the
+// screen.
 @interface InfobarBannerAnimator
     : NSObject <UIViewControllerAnimatedTransitioning>
 
diff --git a/ios/chrome/browser/ui/infobars/presentation/infobar_banner_animator.mm b/ios/chrome/browser/ui/infobars/presentation/infobar_banner_animator.mm
index df8df127e1c82..c49b7b92005c5 100644
--- a/ios/chrome/browser/ui/infobars/presentation/infobar_banner_animator.mm
+++ b/ios/chrome/browser/ui/infobars/presentation/infobar_banner_animator.mm
@@ -15,6 +15,7 @@ - (NSTimeInterval)transitionDuration:
   return 1;
 }
 
+// TODO(crbug.com/911864): PLACEHOLDER animation to present the InfobarBanner.
 - (void)animateTransition:
     (id<UIViewControllerContextTransitioning>)transitionContext {
   // Set up the keys for the "base" view/VC and the "presented" view/VC. These
@@ -72,6 +73,11 @@ - (void)animateTransition:
           [presentedView removeFromSuperview];
         }
 
+        // If dismiss was successful, remove the view.
+        if (!self.presenting && success) {
+          [presentedView removeFromSuperview];
+        }
+
         // Notify UIKit that the transition has finished
         [transitionContext completeTransition:success];
       }];
diff --git a/ios/chrome/browser/ui/infobars/presentation/infobar_banner_presentation_controller.h b/ios/chrome/browser/ui/infobars/presentation/infobar_banner_presentation_controller.h
index bbe80a820d27e..774ee51132254 100644
--- a/ios/chrome/browser/ui/infobars/presentation/infobar_banner_presentation_controller.h
+++ b/ios/chrome/browser/ui/infobars/presentation/infobar_banner_presentation_controller.h
@@ -7,8 +7,7 @@
 
 #import <UIKit/UIKit.h>
 
-// TODO(crbug.com/1372916): PLACEHOLDER Work in Progress class for the new
-// InfobarUI.
+// InfobarBanner Presentation Controller.
 @interface InfobarBannerPresentationController : UIPresentationController
 
 @end
diff --git a/ios/chrome/browser/ui/infobars/presentation/infobar_banner_presentation_controller.mm b/ios/chrome/browser/ui/infobars/presentation/infobar_banner_presentation_controller.mm
index 1bf914ebb02d7..0d37513f489ed 100644
--- a/ios/chrome/browser/ui/infobars/presentation/infobar_banner_presentation_controller.mm
+++ b/ios/chrome/browser/ui/infobars/presentation/infobar_banner_presentation_controller.mm
@@ -26,28 +26,26 @@ @interface InfobarBannerPresentationController ()
 
 @implementation InfobarBannerPresentationController
 
-- (CGRect)frameOfPresentedViewInContainerView {
-  return self.viewForPresentedView.bounds;
-}
-
 - (void)presentationTransitionWillBegin {
   self.containerView.frame = self.viewForPresentedView.frame;
 }
 
 - (void)containerViewWillLayoutSubviews {
-  self.presentedView.frame = [self frameOfPresentedViewInContainerView];
+  self.containerView.frame = self.viewForPresentedView.frame;
+  self.presentedView.frame = self.viewForPresentedView.bounds;
 }
 
+// TODO(crbug.com/911864): PLACEHOLDER position and size for the banner
+// presented view.
 - (UIView*)viewForPresentedView {
-  if (!_viewForPresentedView) {
-    CGFloat safeAreaWidth =
-        CGRectGetWidth(self.containerView.safeAreaLayoutGuide.layoutFrame);
-    CGFloat maxAvailableWidth = safeAreaWidth - 2 * kContainerHorizontalPadding;
-    _viewForPresentedView = [[UIView alloc]
-        initWithFrame:CGRectMake(kContainerHorizontalPadding,
-                                 kContainerTopPadding, maxAvailableWidth,
-                                 kContainerHeight)];
-  }
+  UIWindow* window = UIApplication.sharedApplication.keyWindow;
+  CGFloat safeAreaWidth = CGRectGetWidth(window.bounds);
+  CGFloat maxAvailableWidth = safeAreaWidth - 2 * kContainerHorizontalPadding;
+  _viewForPresentedView = [[UIView alloc]
+      initWithFrame:CGRectMake(kContainerHorizontalPadding,
+                               kContainerTopPadding, maxAvailableWidth,
+                               kContainerHeight)];
+
   return _viewForPresentedView;
 }
 
